record/record.app.js

```diff
@@ -89,6 +89,14 @@
                     var noDataError = ErrorService.noRecordError(context.filter.filters);
                     throw noDataError;
                 }
+                else if(page.tuples.length > 1){
+                    var recordPageLink = page._ref.appLink;
```

Comment:
Since javascript doesn't have a notion of private and public, to define a private property we add `_` before the name of it. That means `_ref` is a private property of Page and you should not use it. You can use `page.reference` instead.

Also don't replace the link, use applink function to get the link for other contexts. In this case:
```javascript
var recordSetLink = page.reference.contextualize.compact.appLink;
```
recordset app is using the `compact` context. That's why you should first contextualize reference to the appropriate context and then get the `appLink` for that.